src/core/utils/scales.ts

```diff
@@ -1,43 +1,32 @@
 import type { GoslingTrackModel } from '../gosling-track-model';
 import type { Assembly, Domain } from '../gosling.schema';
 import { SUPPORTED_CHANNELS } from '../mark';
-import {
-    IsDomainChr,
-    IsDomainInterval,
-    IsDomainChrInterval,
-    IsDomainGene,
-    IsChannelDeep
-} from '../gosling.schema.guards';
+import { IsDomainChr, IsDomainInterval, IsDomainChrInterval, IsChannelDeep } from '../gosling.schema.guards';
 import { GET_CHROM_SIZES } from './assembly';
-import type { Chromosome } from './chrom-size';
 
 /**
  * Get a numeric domain based on a domain specification.
- * For example, domain: { chromosome: '1', interval: [1, 300,000] } => domain: [1, 300,000]
+ * For example, domain: { chromosome: 'chr1', interval: [1, 300,000] } => domain: [1, 300,000]
  */
 export function getNumericDomain(domain: Domain, assembly?: Assembly) {
     if ('chromosome' in domain) {
-        if (domain.chromosome.includes('chr')) {
-            domain.chromosome = domain.chromosome.replace('chr', '') as Chromosome;
-        }
-        if (!Object.keys(GET_CHROM_SIZES(assembly).interval).find(chr => chr === `chr${domain.chromosome}`)) {
-            // we did not find any, so use '1' by default
-            domain.chromosome = '1';
+        const isThereChr = Object.keys(GET_CHROM_SIZES(assembly).interval).find(chr => chr === domain.chromosome);
+        if (!isThereChr) {
+            // Did not find the chromosome, so return early.
+            return;
         }
     }
     if (IsDomainChr(domain)) {
         return [
-            GET_CHROM_SIZES(assembly).interval[`chr${domain.chromosome}`][0] + 1,
-            GET_CHROM_SIZES(assembly).interval[`chr${domain.chromosome}`][1]
+            GET_CHROM_SIZES(assembly).interval[domain.chromosome][0] + 1,
+            GET_CHROM_SIZES(assembly).interval[domain.chromosome][1]
         ];
     } else if (IsDomainInterval(domain)) {
         return domain.interval;
     } else if (IsDomainChrInterval(domain)) {
-        const chrStart = GET_CHROM_SIZES(assembly).interval[`chr${domain.chromosome}`][0];
```

Comment:
I would suggest calling `GET_CHROM_SIZES(assembly)` once in this function and then reusing the returned object. Remember that this function currently creates `URL` objects on every invocation.

I see this pattern a lot in the repo, I think that it stems from the `GET_CHROM_SIZES` name appearing to be a constant (where in fact it is a function that has non-trival overhead of computing). My I suggest renaming `GET_CHROM_SIZES` to `calculateChromsizes` or `computeChromsizes` so that this is more clear?

Reply:
Renaming the function is a good idea. I took the name `computeChromSizes`.